### PR TITLE
Updates pyproject.toml dependency on json_schema

### LINKED ISSUE #5495: Allow JSON Schema to continue forward
### Proposed Changes

Unrestrict jsonschema

### Justification

Patches

### PR DESCRIPTION
**Pull request by jvanderaa**
# Closes #5495
# What's Changed

Removes upper boundary dependency on JSON Schema.

# Screenshots

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by glennmatthews** [comment]
Needs broader discussion about our dependency pinning rather than a point fix, IMHO. Also there still needs to be *some* upper bound, even if only a `<5.0`.

**Comment by jvanderaa** [comment]
I thought the upper bound was controlled by Poetry with having 4.# with the # in there. I will put an upper bound of 5.

**Comment by glennmatthews** [comment]
You also need to use poetry to make these changes so the lock file gets updated as well - never edit dependencies manually in `pyproject.toml`.

**Comment by jvanderaa** [comment]
From the docs this should work to keep in the 4.x - https://python-poetry.org/docs/cli/

```
# Allow >=2.0.5, <3.0.0 versions
poetry add pendulum@^2.0.5
```

**Comment by retnuh** [comment]
Yes `^4.7.0` keeps an upper bound on the major.  TBH most of the places Nautobot is using `~` should probably be `^`.

File: changes/5495.changed [hunk 1]
```diff
@@ -0,0 +1 @@
+Updates pyproject.toml to allow jsonschema to go beyond current version.
```
Comment (gsnider2195) [inline changes/5495.changed:1]: Will also need to rename to `changes/5495.dependencies`

```suggestion
Changed jsonschema version constraint from `>=4.7.0,<4.19.0` to `^4.7.0`.
```

**Review by gsnider2195 — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**